zork++/src/lib/utils/reader.rs

```diff
@@ -329,4 +329,122 @@ mod test {
+
+            [tests]
+            test_executable_name = "zork_check"
+            sources_base_path = "test"
+            sources = [
+                "*.cpp"
+            ]
+            extra_args = [ "-pedantic" ]
+
+            [modules]
+            base_ifcs_dir = "ifc"
+            interfaces = [
+                { filename = "math.cppm" },
+                { filename = 'some_module.cppm', module_name = 'math' }
+            ]
+
+            base_impls_dir = "src"
+            implementations = [
+                { filename = "math.cpp" },
+                { filename = 'some_module_impl.cpp', dependencies = ['iostream'] }
+            ]
+            gcc_sys_headers = [
+                "/usr/include"
+            ]
+        "#;
+
+        let config: ZorkConfigFile = toml::from_str(CONFIG_FILE_MOCK)?;
+        let model = build_model(&config);
+
+        let expected = ZorkModel {
+            project: ProjectModel {
+                name: "Zork++",
+                authors: &["zerodaycode.gz@gmail.com"],
+            },
+            compiler: CompilerModel {
+                cpp_compiler: CppCompiler::CLANG,
+                cpp_standard: LanguageLevel::CPP20,
+                std_lib: Some(StdLib::LIBCPP),
+                extra_args: vec![Argument::from("-Wall")],
+                system_headers_path: Some(Path::new("/usr/include")),
+            },
+            build: BuildModel {
+                output_dir: Path::new("build"),
+            },
+            executable: ExecutableModel {
+                executable_name: "zork",
+                sourceset: SourceSet {
+                    base_path: Path::new("bin"),
+                    sources: vec![Source::Glob(GlobPattern("*.cpp"))],
+                },
+                extra_args: vec![Argument::from("-Werr")],
+            },
+            modules: ModulesModel {
+                base_ifcs_dir: Path::new("ifc"),
+                interfaces: vec![
+                    ModuleInterfaceModel {
+                        filename: Path::new("math.cppm"),
+                        module_name: "math",
+                        dependencies: vec![],
+                    },
+                    ModuleInterfaceModel {
+                        filename: Path::new("some_module.cppm"),
+                        module_name: "math",
+                        dependencies: vec![],
+                    },
+                ],
+                base_impls_dir: Path::new("src"),
+                implementations: vec![
+                    ModuleImplementationModel {
+                        filename: Path::new("math.cpp"),
+                        dependencies: vec!["math"],
+                    },
+                    ModuleImplementationModel {
+                        filename: Path::new("some_module_impl.cpp"),
+                        dependencies: vec!["iostream"],
+                    },
+                ],
+                gcc_sys_headers: vec![Path::new("/usr/include")],
```

Comment:
Just as a side note, this prop doesn't reflects what exactly does. `gcc_sys_headers` should be named `gcc_sys_modules` because `GCC` explicitly requires to build the standard headers as modules, ie:
```
g++ -std=c++20 -fmodules-ts -x c++-system-header iostream
```
if some of the user modules that will be built has a direct dependency of `iostream` (and like that for any other system header`